dspace-api/src/main/java/org/dspace/versioning/VersioningConsumer.java

```diff
@@ -49,37 +73,392 @@ public void finish(Context ctx) throws Exception {
-                if (history != null) {
-                    Version latest = versionHistoryService.getLatestVersion(ctx, history);
-                    Version previous = versionHistoryService.getPrevious(ctx, history, latest);
-                    if (previous != null) {
-                        Item previousItem = previous.getItem();
-                        if (previousItem != null) {
-                            previousItem.setArchived(false);
-                            itemsToProcess.add(previousItem);
-                            //Fire a new modify event for our previous item
-                            //Due to the need to reindex the item in the search
-                            //and browse index we need to fire a new event
-                            ctx.addEvent(new Event(Event.MODIFY,
-                                                   previousItem.getType(), previousItem.getID(),
-                                                   null, itemService.getIdentifiers(ctx, previousItem)));
-                        }
-                    }
+            itemsToProcess = new HashSet<>();
+        }
+
+        // only items
+        if (event.getSubjectType() != Constants.ITEM) {
+            return;
+        }
+
+        // only install events
+        if (event.getEventType() != Event.INSTALL) {
+            return;
+        }
+
+        // get the item (should be archived)
+        Item item = (Item) event.getSubject(ctx);
+        if (item == null || !item.isArchived()) {
+            return;
+        }
+
+        // get version history
+        VersionHistory history = versionHistoryService.findByItem(ctx, item);
+        if (history == null) {
+            return;
+        }
+
+        // get latest version
+        Version latestVersion = versionHistoryService.getLatestVersion(ctx, history);
+        if (latestVersion == null) {
+            return;
+        }
+
+        // get previous version
+        Version previousVersion = versionHistoryService.getPrevious(ctx, history, latestVersion);
+        if (previousVersion == null) {
+            return;
+        }
+
+        // get latest item
+        Item latestItem = latestVersion.getItem();
+        if (latestItem == null) {
+            String msg = String.format(
+                "Illegal state: Obtained version history of item with uuid %s, handle %s, but the latest item is null",
+                item.getID(), item.getHandle()
+            );
+            log.error(msg);
+            throw new IllegalStateException(msg);
+        }
+
+        // get previous item
+        Item previousItem = previousVersion.getItem();
+        if (previousItem == null) {
+            return;
+        }
+
+        // unarchive previous item
+        unarchiveItem(ctx, previousItem);
+
+        // update relationships
+        updateRelationships(ctx, latestItem, previousItem);
+    }
+
+    protected void unarchiveItem(Context ctx, Item item) {
```

Comment:
Remind me again why we unarchive previous versions of Items?  Is this just so they are not returned in search/browse & also to ensure they aren't processed by Curation Tasks or similar?  (We might want to add a brief note here to describe why we do this. I think there's a good reason, and I may have stumbled on why, but it might be good to document it.)

Reply:
We just tried to avoid all changes we can avoid. This comes from https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/versioning/VersioningConsumer.java#L68

We're already dealing with a large amount of unrelated bug fixes in this PR, so we'd prefer to stay away from changing this behavior and risking errors